lightningd/plugin.c

```diff
@@ -49,8 +58,17 @@ struct plugins *plugins_new(const tal_t *ctx, struct log_book *log_book,
 
 static void destroy_plugin(struct plugin *p)
 {
+	struct plugin_rpccall *call;
 	plugin_hook_unregister_all(p);
 	list_del(&p->list);
+
+	/* Terminate all pending RPC calls with an error. */
+	list_for_each(&p->pending_rpccalls, call, list)
+	{
+		was_pending(command_fail(
+		    call->cmd, PLUGIN_PENDING_REQUEST_DIED,
+		    "Plugin terminated before replying to RPC call."));
+	}
```

Comment:
does the p->pending_calls list need to be cleaned up?

Reply:
No, it's allocated off of the plugin, and therefore will be freed as a dependency when we free the `plugin` (which happens to be our signal to die).